lib/libzutil/zutil_import.c

```diff
@@ -929,22 +1006,39 @@ zpool_read_label(int fd, nvlist_t **config, int *num_labels)
 
 	if (lio_listio(LIO_WAIT, aiocbps, VDEV_LABELS, NULL) != 0) {
 		int saved_errno = errno;
-
-		if (errno == EAGAIN || errno == EINTR || errno == EIO) {
-			/*
-			 * A portion of the requests may have been submitted.
-			 * Clean them up.
-			 */
-			for (l = 0; l < VDEV_LABELS; l++) {
-				errno = 0;
-				int r = aio_error(&aiocbs[l]);
-				if (r != EINVAL)
+		errno = 0;
+		int do_slow = 0;
+		error = -1;
+
+		for (l = 0; l < VDEV_LABELS; l++) {
+			switch (aio_error(&aiocbs[l])) {
+				case 0:
+					break;
+				case -1:
+					break;
+				case EOPNOTSUPP:
+					do_slow = 1;
+				case EIO:
+				case EINTR:
+				case EAGAIN:
+					/*
+					 * A portion of the requests may have
+					 * been submitted. Clean them up.
+					 */
 					(void) aio_return(&aiocbs[l]);
 			}
```

Comment:
You need a default statement here.  You should probably `assert` if `aio_error` returns `EINPROGRESS`.  That shouldn't be possible with `LIO_WAIT`.  Or perhaps just print a warning.  In either case, falling back to the slow path might be appropriate for unknown errors.

Reply:
Sure, I opted to ASSERT on it because it looks like printing an error in the way done elsewhere in zutil would require changing the function signature, which seemed like a nonstarter.